mpp/shmem.h.in

```diff
@@ -488,8 +492,313 @@ void *shrealloc(void *ptr, size_t size) __attribute__ ((deprecated));
 void shfree(void *ptr) __attribute__ ((deprecated));
 void start_pes(int npes) __attribute__ ((deprecated));
 
+/* C++ overloaded declarations */
+#ifdef __cplusplus
+} /* extern "C" */
+#undef complex
```

Comment:
Will user code break if we undef complex here?

Can you also add a C++ test case that uses one of the complex routines?  We have C++ bindings and a C implementation.  We should be checking that they mesh correctly with this weird case.